crypto/evp/evp_fetch.c

```diff
@@ -287,17 +288,32 @@ inner_evp_generic_fetch(OPENSSL_CTX *libctx, int operation_id,
     return method;
 }
 
-#ifndef FIPS_MODULE
-static const char *libctx_descriptor(OPENSSL_CTX *libctx)
+/*
+ * Emit an error indicating what the fetch was seeking and where it was
+ * looking.
+ */
+static void generic_fetch_error(ossl_unused OPENSSL_CTX *libctx,
+                                const char *name, const char *properties)
 {
+#ifdef FIPS_MODULE
+    const char *prov_name = "FIPS internal library context";
```

Comment:
This suggests something I've had in the back of my mind, that we should name the library contexts...

Reply:
Me too ... with a named library context we get cross context functionality.  Pass the digest, property query and library context name to HMAC or digest sign instead of just the first two.

The additional name here isn't the important point, the first stab at it didn't name bother.  The algorithm and property queries are the important diagnostic.  Once I started refactoring common code, the name made sense.